homeassistant/components/automation/__init__.py

```diff
@@ -458,15 +458,19 @@ async def async_trigger(self, run_variables, context=None, skip_condition=False)
             trigger_context,
             self._trace_config,
         ) as automation_trace:
+
+            self.async_write_ha_state()
```

Comment:
Is this really needed here, it should only be needed in `_async_attach_triggers`?

Reply:
I will remove and retest.  I copied the stragegy from _async_attach_triggers ... but it should already have it's state set before  the trigger.